site-src/geps/gep-724.md

```diff
@@ -0,0 +1,624 @@
+  // +kubebuilder:default=Gateway
+  Kind string `json:"kind"`
+
+  // Namespace is the namespace of the referent. When unspecified (empty
+  // string), this will either be:
+  //
+  // * local namespace of the target is a namespace scoped resource
+  // * no namespace (not applicable) if the target is cluster-scoped.
+  //
+  // Support: Extended
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  Namespace string `json:"namespace,omitempty"`
+
+  // Scope represents if this refers to a cluster or namespace scoped resource.
+  // This may be set to "Cluster" or "Namespace".
+  //
+  // Support: Core (Namespace)
+  // Support: Extended (Cluster)
+  //
+	// +kubebuilder:validation:Enum=Cluster;Namespace
+  // +kubebuilder:default=Namespace
+  // +optional
+  Scope string `json:"scope,omitempty"`
+
+  // Name is the name of the referent.
+  //
+  // Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Name string `json:"name"`
+
+  // SectionName is the name of a section within the target resource. In the
+  // following resources, SectionName is interpreted as the following:
+  //
+  // * Gateway: Listener Name
+  //
+  // Implementations MAY choose to support attaching Routes to other resources.
+  // If that is the case, they MUST clearly document how SectionName is
+  // interpreted.
+  //
+  // When unspecified (empty string), this will reference the entire resource.
+  // For the purpose of status, an attachment is considered successful if at
+  // least one section in the parent resource accepts it. For example, Gateway
+  // listeners can restrict which Routes can bind to them by Route kind,
+  // namespace, or hostname. If 1 of 2 Gateway listeners accept attachment from
+  // the referencing Route, the Route MUST be considered successfully
+  // attached. If no Gateway listeners accept attachment from this Route, the
+  // Route MUST be considered detached from the Gateway.
+  //
+  // Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  SectionName string `json:"sectionName,omitempty"`
+}
+```
+
+This would also require adding a `Name` field to Gateway listeners and Route
+rules to support the SectionName concept:
+
+```go
+type Listener struct {
+    // Name is the name of the Listener. If more than one Listener is present
+    // each Listener MUST specify a name. The names of Listeners MUST be unique
+    // within a Gateway.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    Name string `json:"name,omitempty"`
+    // ...
+}
+```
+
+#### Changed
+
+To accomodate Routes with arbitrary types of parents, `RouteGatewayStatus` would
+be renamed to `RouteParentStatus`. Similarly, the `GatewayRef` inside that
+struct would be
```

Comment:
The sentence is unterminated and the thought incomplete.

Reply:
🤦 woops! Should be updated now.